src/Recruit.Vacancies.Client/Domain/Entities/Vacancy.cs

```diff
@@ -17,6 +17,17 @@ public class Vacancy
 
         public DateTime? SubmittedDate { get; set; }
 
-        public bool CanSubmit => Status == VacancyStatus.Draft;
+        public bool IsDeleted { get; set; }
+        public DateTime? DeletedDate { get; set; }
+
+        /// <summary>
+        /// We can only submit draft vacancies that have not been deleted
+        /// </summary>
+        public bool CanSubmit => Status == VacancyStatus.Draft && IsDeleted == false;
```

Comment:
Do these field get persisted to the Vacancy in Mongo/Cosmos as well?

Reply:
Nope. Just checked